packages/playwright-core/src/server/browserContext.ts

```diff
@@ -122,6 +122,20 @@ export abstract class BrowserContext extends SdkObject {
 
     if (debugMode() === 'console')
       await this.extendInjectedScript(consoleApiSource.source);
+    if (this._options.serviceWorkerPolicy === 'disabled') {
+      await this.addInitScript(`
+
+        // Service Worker Policy Injected Script
+
+        navigator.serviceWorker.register = () => {
+          console.warn('Service Worker registration disabled by Playwright');
+          return Promise.reject('Service Worker registration disabled by Playwright');
```

Comment:
I would probably stall this one.

Reply:
For stall here, do you mean return (A) a Promise that never resolves or (B) immediately return a resolved promise?